homeassistant/components/nest/camera_sdm.py

```diff
@@ -90,11 +96,24 @@ async def stream_source(self):
         if CameraLiveStreamTrait.NAME not in self._device.traits:
             return None
         trait = self._device.traits[CameraLiveStreamTrait.NAME]
-        rtsp_stream = await trait.generate_rtsp_stream()
+        now = utcnow()
+        if not self._stream:
+            logging.debug("Fetching stream url")
+            self._stream = await trait.generate_rtsp_stream()
+        elif self._stream.expires_at < now:
+            logging.debug("Stream expired, extending stream")
+            new_stream = await self._stream.extend_rtsp_stream()
+            self._stream = new_stream
         # Note: This is only valid for a few minutes, and probably needs
         # to be improved with an occasional call to .extend_rtsp_stream() which
         # returns a new rtsp_stream object.
-        return rtsp_stream.rtsp_stream_url
+        return self._stream.rtsp_stream_url
+
+    async def async_removed_from_registry(self):
```

Comment:
You should do it anytime this entity is unloaded, not just when it's completely removed:

```suggestion
    async def async_will_remove_from_hass(self):
```